internal/provider/resource_cloudflare_account.go

```diff
@@ -0,0 +1,150 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	accountTypeStandard   = "standard"
+	accountTypeEnterprise = "enterprise"
+)
+
+func resourceCloudflareAccount() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareAccountSchema(),
+		CreateContext: resourceCloudflareAccountCreate,
+		ReadContext:   resourceCloudflareAccountRead,
+		UpdateContext: resourceCloudflareAccountUpdate,
+		DeleteContext: resourceCloudflareAccountDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: heredoc.Doc(`
+			Provides a Cloudflare Account resource. Account is the basic resource for
+			working with Cloudflare zones, teams and users. Requires the tenant entitlement
+		`),
+	}
+}
+
+func resourceCloudflareAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountName := d.Get("name").(string)
+	var accountType string
+	if d.Get("type").(string) == "" {
```

Comment:
we can do this logic with a combination of `Default: "standard"` and `validation.StringInSlice` (example: https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/internal/provider/schema_cloudflare_list.go#L23) for confirming only allowed values.

Reply:
Agreed. It was already checked at schema level this way so I removed this useless check here.